### ISSUE TITLE
Plain docker mode: port mapping doesn't seem to work

### ISSUE THREAD

**Issue by dustymabe** [minishift/minishift#588]
*note* this was on cdk 3 beta. I have not yet had a chance to try on centos version

when I run `minishift docker-env` and then try to use docker to run a container and map ports I get an error:

```
[cloud-user@cloudhost ~]$ docker run -dt -p 9000:80 fedora/apache
259887c99b07e71402e1692bf84b46293d4ba8b46ce44c20335001060e63e459
/usr/bin/docker-current: Error response from daemon: driver failed programming external connectivity on endpoint elated_ritchie (ed3c8d3fa6a89262a888af108661d49d78300f09b5d58f9697ce60c8cc5ce1bc): exec: "docker-proxy": executable file not found in $PATH.

[cloud-user@cloudhost ~]$ minishift version
Minishift version: 1.0.0-beta.4-1
CDK Version: 3.0.0-beta.2

```

**Comment by gbraad** [comment]
Noticed the same when trying to run Che in the docker environment [ref](https://gist.github.com/gbraad/ac1be4bcbd47513acf1aa4a56e28094e) Which fails for the same reason.

Was investigating this...

**Comment by dustymabe** [comment]
running this in the VM (i.e. adding it to the ISO) allows me to workaround this issue:

```
sudo ln -s ../../usr/libexec/docker/docker-proxy-current /usr/bin/docker-proxy
```

**Comment by dustymabe** [comment]
actually, this turns out to be a misconfiguration:

```
[docker@minishift ~]$ cat /etc/systemd/system/docker.service | grep proxy
[docker@minishift ~]$ cat /usr/lib/systemd/system/docker.service | grep proxy
          --userland-proxy-path=/usr/libexec/docker/docker-proxy-current \
[docker@minishift ~]$ rpm -q docker
docker-1.12.5-14.el7.x86_64
```

we should probably fine a way to have a file that "adds in the settings that we want for minishift" but is also able to pick up changes from the docker rpm when they get delivered.

**Comment by gbraad** [comment]
@praveenkumar since the configuration is not persistent (or gets overwritten), we could at least make the symbolic link from the `handle-user-data`.

**Comment by praveenkumar** [comment]
@gbraad I agree we need to find a better way to solve it. Current setting we used because it's easy for us to set docker-env, storage options etc... when user defined.

```
[Unit]
Description=Docker Application Container Engine
After=network.target rc-local.service
Requires=rhel-push-plugin.socket
[Service]
Type=notify
ExecStart=/usr/bin/docker daemon -H tcp://0.0.0.0:{{.DockerPort}} -H unix:///var/run/docker.sock \
           --authorization-plugin rhel-push-plugin \
           --selinux-enabled \
           --add-runtime docker-runc=/usr/libexec/docker/docker-runc-current \
           --default-runtime=docker-runc \
           --add-registry registry.access.redhat.com \
           --storage-driver {{.EngineOptions.StorageDriver}} --tlsverify --tlscacert {{.AuthOptions.CaCertRemotePath}} \
           --tlscert {{.AuthOptions.ServerCertRemotePath}} --tlskey {{.AuthOptions.ServerKeyRemotePath}} \
           {{ range .EngineOptions.Labels }}--label {{.}} {{ end }}{{ range .EngineOptions.InsecureRegistry }}--insecure-registry {{.}} {{ end }}{{ range .EngineOptions.RegistryMirror }}--registry-mirror {{.}} {{ end }}{{ range .EngineOptions.ArbitraryFlags }}--{{.}} {{ end }}
ExecReload=/bin/kill -s HUP $MAINPID
MountFlags=slave
LimitNOFILE=infinity
LimitNPROC=infinity
LimitCORE=infinity
TimeoutStartSec=0
Delegate=yes
KillMode=process
Environment={{range .EngineOptions.Env}}{{ printf "%q" . }} {{end}}
[Install]
WantedBy=multi-user.target
```

**Comment by praveenkumar** [comment]
We should start using something https://coreos.com/os/docs/latest/using-systemd-drop-in-units.html

**Comment by hferentschik** [comment]
Resolved via pull request #605
